app/src/main/java/com/keylesspalace/tusky/components/trending/TrendingLinksFragment.kt

```diff
@@ -0,0 +1,233 @@
+                        } else {
+                            if (it.throwable !is HttpException || it.throwable.code() != 404) {
+                                binding.messageView.setup(it.throwable) {
+                                    viewModel.accept(InfallibleUiAction.Reload)
+                                }
+                            } else {
+                                binding.messageView.setup(it.throwable)
+                            }
+                            binding.messageView.show()
+                        }
+                    }
+                }
+            }
+        }
+
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewModel.statusDisplayOptions.collectLatest {
+                adapter.statusDisplayOptions = it
+            }
+        }
+
+        (activity as? ActionButtonActivity)?.actionButton?.hide()
+    }
+
+    private fun setupSwipeRefreshLayout() {
+        binding.swipeRefreshLayout.setOnRefreshListener(this)
+        binding.swipeRefreshLayout.setColorSchemeResources(R.color.tusky_blue)
+    }
+
+    private fun setupRecyclerView() {
+        binding.recyclerView.layoutManager = getLayoutManager(requireContext().resources.getInteger(R.integer.trending_column_count))
+        binding.recyclerView.setHasFixedSize(true)
+        (binding.recyclerView.itemAnimator as SimpleItemAnimator).supportsChangeAnimations = false
+        binding.recyclerView.adapter = adapter
+    }
+
+    private fun getLayoutManager(columnCount: Int) = GridLayoutManager(context, columnCount)
+
+    override fun onCreateMenu(menu: Menu, menuInflater: MenuInflater) {
+        menuInflater.inflate(R.menu.fragment_trending_links, menu)
+        menu.findItem(R.id.action_refresh)?.apply {
+            icon = IconicsDrawable(requireContext(), GoogleMaterial.Icon.gmd_refresh).apply {
+                sizeDp = 20
+                colorInt =
+                    MaterialColors.getColor(binding.root, android.R.attr.textColorPrimary)
+            }
+        }
+    }
+
+    override fun onMenuItemSelected(menuItem: MenuItem): Boolean {
+        return when (menuItem.itemId) {
+            R.id.action_refresh -> {
+                refreshContent()
+                true
+            }
+            else -> false
+        }
+    }
+
+    override fun refreshContent() {
+        binding.swipeRefreshLayout.isRefreshing = true
+        onRefresh()
+    }
+
+    override fun onRefresh() = viewModel.accept(InfallibleUiAction.Reload)
+
+    override fun onReselect() {
+        if (isAdded) {
+            binding.recyclerView.layoutManager?.scrollToPosition(0)
+            binding.recyclerView.stopScroll()
+        }
+    }
+
+    private fun onOpenLink(url: String) = requireContext().openLink(url)
+
+    override fun onResume() {
+        super.onResume()
+        val a11yManager =
+            ContextCompat.getSystemService(requireContext(), AccessibilityManager::class.java)
+
+        val wasEnabled = talkBackWasEnabled
```

Comment:
This is only done for the trending fragments?

(Looks like a job for a base class - and then used elsewhere, too?)

Reply:
No, it's done in `TimelineFragment` as well, so anything that inherits from that gets it too.

Before you ask, this can't inherit from `TimelineFragment`, because that's written to assume that it's showing a timeline of statuses, not a timeline of arbitrary other items.

Reply:
(can be resolved)